diff -up ./src/htsmsg.c.orig ./src/htsmsg.c
--- ./src/htsmsg.c.orig	2016-03-14 10:10:57.000000000 +0100
+++ ./src/htsmsg.c	2016-07-05 00:38:18.670529721 +0200
@@ -760,13 +760,15 @@ htsmsg_print0(htsmsg_t *msg, int indent)
     case HMF_MAP:
       printf("MAP) = {\n");
       htsmsg_print0(&f->hmf_msg, indent + 1);
-      for(i = 0; i < indent; i++) printf("\t"); printf("}\n");
+      for(i = 0; i < indent; i++) printf("\t");
+      printf("}\n");
       break;
 
     case HMF_LIST:
       printf("LIST) = {\n");
       htsmsg_print0(&f->hmf_msg, indent + 1);
-      for(i = 0; i < indent; i++) printf("\t"); printf("}\n");
+      for(i = 0; i < indent; i++) printf("\t");
+      printf("}\n");
       break;
       
     case HMF_STR:
