diff -up ./rpm/tvheadend.sysconfig.orig ./rpm/tvheadend.sysconfig
--- ./rpm/tvheadend.sysconfig.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./rpm/tvheadend.sysconfig	2017-04-23 13:24:58.581096686 +0200
@@ -1,4 +1,4 @@
 # Configuration file for the tvheadend service.
 
 MALLOC_ARENA_MAX=4
-OPTIONS="-c /home/tvheadend/config -u tvheadend -g tvheadend -6 --http_port 9981 --htsp_port 9982"
+OPTIONS="-c /var/lib/tvheadend/config -u tvheadend -g video -6 -C --http_port 9981 --htsp_port 9982"
