diff -up ./configure.orig ./configure
--- ./configure.orig	2016-07-05 01:42:43.615016618 +0200
+++ ./configure	2016-07-05 01:44:12.239677665 +0200
@@ -290,7 +290,7 @@ if enabled hdhomerun_static; then
 else
 
   if enabled_or_auto hdhomerun_client; then
-    if check_cc_header 'libhdhomerun/hdhomerun' libhdhomerun; then
+    if check_cc_header 'hdhomerun/hdhomerun' libhdhomerun; then
       enable  hdhomerun_client
       LDFLAGS="$LDFLAGS -lhdhomerun"
     fi
diff -up ./src/input/mpegts/tvhdhomerun/tvhdhomerun_private.h.orig ./src/input/mpegts/tvhdhomerun/tvhdhomerun_private.h
--- ./src/input/mpegts/tvhdhomerun/tvhdhomerun_private.h.orig	2016-03-14 10:10:57.000000000 +0100
+++ ./src/input/mpegts/tvhdhomerun/tvhdhomerun_private.h	2016-07-05 01:44:32.826134351 +0200
@@ -24,7 +24,7 @@
 #include "htsbuf.h"
 #include "tvhdhomerun.h"
 
-#include <libhdhomerun/hdhomerun.h>
+#include <hdhomerun/hdhomerun.h>
 
 typedef struct tvhdhomerun_device_info tvhdhomerun_device_info_t;
 typedef struct tvhdhomerun_device      tvhdhomerun_device_t;
