diff -up ./src/input/mpegts/scanfile.c.orig ./src/input/mpegts/scanfile.c
--- ./src/input/mpegts/scanfile.c.orig	2021-06-12 15:56:23.000000000 +0200
+++ ./src/input/mpegts/scanfile.c	2021-07-09 16:25:21.695025654 +0200
@@ -921,7 +921,7 @@ scanfile_init ( const char *muxconf_path
 #elif defined(PLATFORM_FREEBSD)
     path = "/usr/local/share/dtv-scan-tables";
 #else
-    path = "/usr/share/dvb";
+    path = "/usr/share/dvbv5";
 #endif
 
   if (!initialized) {
