diff -up ./configure.orig ./configure
--- ./configure.orig	2017-04-23 13:17:04.634688197 +0200
+++ ./configure	2017-04-23 13:18:15.837805616 +0200
@@ -673,6 +673,8 @@ fi
 #
 if enabled_or_auto libsystemd_daemon; then
   if check_pkg libsystemd-daemon; then
+      enable libsystemd_daemon
+  elif check_pkg libsystemd; then
     enable libsystemd_daemon
   elif enabled libsystemd_daemon; then
     die "libsystemd-daemon development support not found (use --disable-systemd_daemon)"
