diff -up ./src/dvr/dvr_rec.c.orig ./src/dvr/dvr_rec.c
--- ./src/dvr/dvr_rec.c.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./src/dvr/dvr_rec.c	2017-04-23 14:05:36.028066833 +0200
@@ -613,7 +613,7 @@ dvr_sub_basic_info(const char *id, const
   HTSMSG_FOREACH(f, info) {
     if (!(e = htsmsg_field_get_map(f))) continue;
     if ((s = htsmsg_get_str(e, "type")) != NULL) continue;
-    tvh_strlcatf(tmp, tmplen, l, "%s%s", l > 0 ? "," : "", s);
+    tvh_strlcatf(tmp, tmplen, l, "%s%s", l > 0 ? "," : "", s == NULL ? "" : s);
   }
   return tmp;
 }
@@ -915,7 +915,7 @@ dvr_rec_start(dvr_entry_t *de, const str
   const source_info_t *si = &ss->ss_si;
   streaming_start_t *ss_copy;
   const streaming_start_component_t *ssc;
-  char res[11], asp[6], sr[6], ch[7];
+  char res[14], asp[14], sr[6], ch[7];
   dvr_config_t *cfg = de->de_config;
   profile_chain_t *prch = de->de_chain;
   htsmsg_t *info, *e;
diff -up ./src/epggrab/module/xmltv.c.orig ./src/epggrab/module/xmltv.c
--- ./src/epggrab/module/xmltv.c.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./src/epggrab/module/xmltv.c	2017-04-23 13:43:54.253923983 +0200
@@ -843,7 +843,7 @@ static void _xmltv_load_grabbers ( void
   /* Internal search */
   } else if ((tmp = getenv("PATH"))) {
     tvhdebug(LS_XMLTV, "using internal grab search");
-    char bin[256];
+    char bin[257];
     char *argv[] = {
       NULL,
       (char *)"--description",
diff -up ./src/lock.c.orig ./src/lock.c
--- ./src/lock.c.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./src/lock.c	2017-04-23 13:42:15.592554853 +0200
@@ -31,7 +31,7 @@ int state_lock_(const char *lfile, int l
     int fd = -1, err = 0;
     struct flock lck;
     struct stat st;
-    char lcktxt[12];
+    char lcktxt[13];
 
     lck.l_type = lock ? F_WRLCK : F_UNLCK;
     lck.l_whence = SEEK_SET;
diff -up ./src/subscriptions.c.orig ./src/subscriptions.c
--- ./src/subscriptions.c.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./src/subscriptions.c	2017-04-23 13:41:50.709217734 +0200
@@ -976,7 +976,7 @@ subscription_create_msg(th_subscription_
   descramble_info_t *di;
   service_t *t;
   profile_t *pro;
-  char buf[256];
+  char buf[284];
   const char *state;
 
   htsmsg_add_u32(m, "id", s->ths_id);
diff -up ./src/webui/webui.c.orig ./src/webui/webui.c
--- ./src/webui/webui.c.orig	2017-04-21 10:32:22.000000000 +0200
+++ ./src/webui/webui.c	2017-04-23 14:07:05.205688547 +0200
@@ -174,7 +174,7 @@ page_login(http_connection_t *hc, const
 {
   if (hc->hc_access != NULL &&
       hc->hc_access->aa_username != NULL &&
-      hc->hc_access->aa_username != '\0') {
+      *(hc->hc_access->aa_username) != '\0') {
     http_redirect(hc, "/", &hc->hc_req_args, 0);
     return 0;
   } else {
@@ -187,7 +187,7 @@ page_logout(http_connection_t *hc, const
 {
   if (hc->hc_access == NULL ||
       hc->hc_access->aa_username == NULL ||
-      hc->hc_access->aa_username == '\0') {
+      *(hc->hc_access->aa_username) == '\0') {
 redirect:
     http_redirect(hc, "/", &hc->hc_req_args, 0);
     return 0;
