diff -up ./src/plumbing/transcoding.c.orig ./src/plumbing/transcoding.c
--- ./src/plumbing/transcoding.c.orig	2016-08-09 12:03:40.771743804 +0200
+++ ./src/plumbing/transcoding.c	2016-08-09 12:06:47.919483730 +0200
@@ -716,12 +716,12 @@ transcoder_stream_audio(transcoder_t *t,
       // Convert audio
       tvhtrace("transcode", "%04X: converting audio", shortid(t));
 
-      tvhtrace("transcode", "%04X: IN : channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%ld",
+      tvhtrace("transcode", "%04X: IN : channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%"PRId64,
                shortid(t), ictx->channels, ictx->channel_layout, ictx->sample_rate,
-               ictx->sample_fmt, ictx->bit_rate);
-      tvhtrace("transcode", "%04X: OUT: channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%ld",
+               ictx->sample_fmt, (int64_t)ictx->bit_rate);
+      tvhtrace("transcode", "%04X: OUT: channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%"PRId64,
                shortid(t), octx->channels, octx->channel_layout, octx->sample_rate,
-               octx->sample_fmt, octx->bit_rate);
+               octx->sample_fmt, (int64_t)octx->bit_rate);
 
       if (transcode_opt_set_int(t, ts, as->resample_context,
                                 "in_channel_layout", ictx->channel_layout, 1))
